src/types/instantsearch.ts

```diff
@@ -1,25 +1,12 @@
-import { EventEmitter } from 'events';
 import { Client as AlgoliaSearchClient } from 'algoliasearch';
-import {
-  AlgoliaSearchHelper,
-  SearchParameters as AlgoliaSearchHelperSearchParameters,
-  SearchResults as AlgoliaSearchHelperSearchResults,
-} from 'algoliasearch-helper';
+import { AlgoliaSearchHelper, SearchParameters } from 'algoliasearch-helper';
 import { Index } from '../widgets/index/index';
 import { InsightsClient as AlgoliaInsightsClient } from './insights';
 import { Widget, UiState } from './widget';
 
 export type InstantSearchOptions = any;
 
-// That's a proxy to avoid manipulating the original `algoliasearch-helper`
-// SearchParameters typings and to add newer search parameters not yet
-// documented or wrongly typed.
-export type SearchParameters = AlgoliaSearchHelperSearchParameters & {
-  ruleContexts?: string[];
-};
-
-export type SearchResults = AlgoliaSearchHelperSearchResults;
-
+// @TODO: can this be written some other way?
```

Comment:
WDYM?

Reply:
I am wondering if it could be extracted from the helper's types, but it isn't really essential

Reply:
Yes it would be interesting rather than inlining the types inside the listeners.